data/items.ts

```diff
@@ -1420,20 +1420,43 @@ export const Items: {[itemid: string]: ItemData} = {
 		},
 		onAfterBoost(boost, target, source, effect) {
 			if (this.activeMove?.id === 'partingshot') return;
-			let eject = false;
+			target.abilityState.eject = false;
 			let i: BoostID;
 			for (i in boost) {
 				if (boost[i]! < 0) {
-					eject = true;
+					target.abilityState.eject = true;
 				}
 			}
-			if (eject) {
+			if (target.abilityState.eject) {
 				if (target.hp) {
 					if (!this.canSwitch(target.side)) return;
-					for (const pokemon of this.getAllActive()) {
-						if (pokemon.switchFlag === true) return;
+					let lastToMove = true;
```

Comment:
Not sure if this is the best way to do it. My approach is not switching in OnAfterBoost if you were the last one to move, in order to allow some end of turn effects like Power Construct to happen before the switch. I'm using this check to switch on OnResidual if that's the case